lib/compat/wordpress-6.1/template-parts-screen.php

```diff
@@ -0,0 +1,208 @@
+		$redirect_url = add_query_arg(
+			array( 'postType' => 'wp_template_part' ),
+			admin_url( 'themes.php?page=gutenberg-template-parts' )
+		);
+		wp_safe_redirect( $redirect_url );
+		exit;
+	}
+}
+add_action( 'load-appearance_page_gutenberg-template-parts', 'gutenberg_template_parts_screen_permissions' );
+
+/**
+ * Initialize the editor for the screen. Most of this is copied from `site-editor.php`.
+ *
+ * Note: Parts that need to be ported back should have inline comments.
+ *
+ * @param string $hook Current page hook.
+ * @return void
+ */
+function gutenberg_template_parts_screen_init( $hook ) {
+	global $current_screen, $editor_styles;
+
+	if ( 'appearance_page_gutenberg-template-parts' !== $hook ) {
+		return;
+	}
+
+	// Flag that we're loading the block editor.
+	$current_screen->is_block_editor( true );
+
+	// Default to is-fullscreen-mode to avoid jumps in the UI.
+	add_filter(
+		'admin_body_class',
+		static function( $classes ) {
+			return "$classes is-fullscreen-mode";
+		}
+	);
+
+	$indexed_template_types = array();
+	foreach ( get_default_block_template_types() as $slug => $template_type ) {
+		$template_type['slug']    = (string) $slug;
+		$indexed_template_types[] = $template_type;
+	}
+
+	$block_editor_context = new WP_Block_Editor_Context( array( 'name' => 'core/edit-site' ) );
+	$custom_settings      = array(
+		'siteUrl'                   => site_url(),
+		'postsPerPage'              => get_option( 'posts_per_page' ),
+		'styles'                    => get_block_editor_theme_styles(),
+		'defaultTemplateTypes'      => $indexed_template_types,
+		'defaultTemplatePartAreas'  => get_allowed_block_template_part_areas(),
+		'supportsLayout'            => WP_Theme_JSON_Resolver::theme_has_support(),
+		'supportsTemplatePartsMode' => ! wp_is_block_theme() && current_theme_supports( 'block-template-parts' ),
+		'__unstableHomeTemplate'    => gutenberg_resolve_home_template(),
+	);
+
+	/**
+	 * We don't need home template resolution when block template parts are supported.
+	 * Set the value to true to satisfy the editor initialization guard clause.
+	 */
+	if ( $custom_settings['supportsTemplatePartsMode'] ) {
+		$custom_settings['__unstableHomeTemplate'] = true;
+	}
+
+	// Add additional back-compat patterns registered by `current_screen` et al.
+	$custom_settings['__experimentalAdditionalBlockPatterns']          = WP_Block_Patterns_Registry::get_instance()->get_all_registered( true );
+	$custom_settings['__experimentalAdditionalBlockPatternCategories'] = WP_Block_Pattern_Categories_Registry::get_instance()->get_all_registered( true );
+
+	$editor_settings = get_block_editor_settings( $custom_settings, $block_editor_context );
+
+	if ( isset( $_GET['postType'] ) && ! isset( $_GET['postId'] ) ) {
+		$post_type = get_post_type_object( $_GET['postType'] );
+		if ( ! $post_type ) {
+			wp_die( __( 'Invalid post type.', 'gutenberg' ) );
+		}
+	}
+
+	$active_global_styles_id = WP_Theme_JSON_Resolver::get_user_global_styles_post_id();
```

Comment:
@Mamaduka @scruffian @fabiankaegy @Soean We should use the `WP_Theme_JSON_Resolver_Gutenberg` class instead of the core one. Fix for this at https://github.com/WordPress/gutenberg/pull/49198